src/java/htsjdk/samtools/sra/SRAAccession.java

```diff
@@ -29,11 +29,19 @@
 import htsjdk.samtools.util.Log;
 import gov.nih.nlm.ncbi.ngs.NGS;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.io.Serializable;
+import java.util.Arrays;
 
 /**
- * Describes a single SRA accession
+ * Describes a single SRA accession for SRA read collection
  * Also provides app string functionality and allows to check if working SRA is supported on the running platform
+ *
+ * Important: due to checks performed in SRAAccession.isValid(), we won't recognise any accessions other
+ * than ones that follow the pattern "^[SED]RR[0-9]{6,9}$"
  */
 public class SRAAccession implements Serializable {
```

Comment:
Since this method now has somewhat non-trivial logic in it, could you add a unit test for it.